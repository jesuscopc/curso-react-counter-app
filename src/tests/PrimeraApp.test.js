import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";

describe('Suit - test <PrimeraApp/>>', () => {
  // test('Debe mostrar el mensaje "Hola. soy Goku', () => {
  //   const saludo = 'Hola, soy Goku';
  //   const wrapper = render(<PrimeraApp  saludo={saludo}/>);
  //   expect(wrapper.getByText(saludo)).toBeInTheDocument();
  // }) 

  test('should be show <PrimeraApp/> correctly', () => {
    const saludo = 'Hola, soy Goku';
    const wrapper = shallow(<PrimeraApp saludo={saludo}/>);
    expect(wrapper).toMatchSnapshot();
  })
  
  test('should be render a custom subtitle', () => {
    const saludo = 'Hola, soy Goku';
    const subtitulo = 'Hi I am Subtitle';
    const wrapper = shallow(
      <PrimeraApp 
        saludo={saludo}
        subtitulo={subtitulo}
      />)
    const textoP = wrapper.find('p').text();
    expect(textoP).toBe(subtitulo); 
  })
  
})
