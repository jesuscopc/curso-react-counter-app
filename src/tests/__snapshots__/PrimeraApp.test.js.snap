// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suit - test <PrimeraApp/>> should be show <PrimeraApp/> correctly 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Soy un Subtitulo
  </p>
</Fragment>
`;
